Modules/Core/Common/include/itkThreadPool.h

```diff
@@ -124,13 +124,23 @@ auto result = pool->AddWork([](int param) { return param; }, 7);
   SetDoNotWaitForThreads(bool doNotWaitForThreads);
 
 protected:
-  /* We need access to the mutex in AddWork, and the variable is only
+  /** We need access to the mutex in AddWork, and the variable is only
    * visible in .cxx file, so this method returns it. */
   std::mutex &
   GetMutex();
 
   ThreadPool();
-  ~ThreadPool() override;
+
+  /** Stop the pool and release threads. To be called by the destructor and atfork. */
+  void
+  CleanUp();
+
+  ~ThreadPool() override { this->CleanUp(); }
+
+  static void
+  atfork_prepare();
```

Comment:
Per style guidelines: `atfork_prepare` -> `PrepareForFork`

Reply:
Sure, I can do that. Do you have any advice about overcoming `multiprocessing.pool.MaybeEncodingError: Error sending result: '[<itk.itkImagePython.itkImageUC3; proxy of <Swig Object of type 'itkImageUC3 *' at 0x7f6ee94955a0> >]'. Reason: 'TypeError("cannot pickle 'SwigPyObject' object")'`? This is when invoking the [repro script](https://github.com/InsightSoftwareConsortium/ITK/issues/2069#issuecomment-754122517).

Reply:
That issue is discussed in #1091 #1948